src/utils/date-utils.js

```diff
@@ -57,3 +57,21 @@ export const getMonthNameFromNumber = (monthNumber) => {
 
   return months[parsedMonthNumber - 1]
 }
+
+/**
+ * Extracts the date portion (YYYY-MM-DD) from an ISO 8601 date-time string
+ *
+ * @param {string} isoDateTime - The ISO 8601 date-time string to extract the date from, e.g. 2024-08-02T00:00:00+01:00
+ * @returns {string} - The extracted date in YYYY-MM-DD format
+ * @throws {Error} - If the input is not a valid ISO 8601 date-time string
+ */
+export const extractDateFromISO = (isoDateTime) => {
+  if (
+    !isoDateTime ||
+    typeof isoDateTime !== 'string' ||
+    !isoDateTime.includes('T')
+  ) {
+    throw new Error('Invalid ISO 8601 date-time string')
+  }
+  return isoDateTime.split('T')[0]
+}
```

Comment:
Rather than doing this, I think you'd be better off using the Date object, so:
```
const extractedDate = new Date(isoDateTime)
if (Number.isNaN(extractedDate.getTime())) {
  throw new Error('Invalid ISO 8601 date-time string')
}
return `${extractedDate.getFullYear()}-${(extractedDate.getMonth() + 1).toString().padStart(2, '0')}-${extractedDate.getDate().toString().padStart(2, '0')}`
```